sweagent/agent/models.py

```diff
@@ -220,7 +223,11 @@ def __init__(self, args: ModelArguments, commands: list[Command]):
 
         # Set OpenAI key
         cfg = config.Config(os.path.join(os.getcwd(), "keys.cfg"))
-        self.client = OpenAI(api_key=cfg["OPENAI_API_KEY"])
+        if self.args.model_name.startswith("azure"):
+             self.api_model=cfg["AZURE_OPENAI_DEPLOYMENT"]
+             self.client = AzureOpenAI(api_key=cfg["AZURE_OPENAI_API_KEY"], azure_endpoint=cfg["AZURE_OPENAI_ENDPOINT"], api_version="2024-02-01")
```

Comment:
How about parametrizing the `api_version` too? It changes rather frequently :)
```suggestion
             self.api_model = cfg["AZURE_OPENAI_DEPLOYMENT"]
             api_version = cfg["OPENAI_API_VERSION"] or "2024-02-01"
             self.client = AzureOpenAI(api_key=cfg["AZURE_OPENAI_API_KEY"], azure_endpoint=cfg["AZURE_OPENAI_ENDPOINT"], api_version=api_version)
```

Reply:
we could make it a `.get("...", default value)` to get the best of both worlds

Reply:
Suggestion updated :)

Reply:
Thanks! Could you check that this works? (I don't have azure credits atm)